test/Chainweb/Test/Utils.hs

```diff
@@ -225,6 +238,18 @@ import P2P.Node.Configuration
 import qualified P2P.Node.PeerDB as P2P
 import P2P.Peer
 
+-- -------------------------------------------------------------------------- --
+-- Misc
+
+#if !MIN_VERSION_base(4,15,0)
+-- | Guarantee that the
+readFile' :: FilePath -> IO String
+readFile' fp = withFile fp ReadMode $ \h -> do
+    s <- hGetContents h
+    void $ evaluate $ length s
+    return s
+#endif
```

Comment:
`readFile'` was added to `base` in `4.15`. For earlier versions of base the code used the implementation from the `extra` package. Once we upgrade to ghc-9, this code can be removed.